### ISSUE TITLE
Reverse Audit: Components IN paper but MISSING/WRONG in code

### ISSUE THREAD

**Issue by toddwbucy** [toddwbucy/Atlas-MAG_OmegaRule#37]
## Summary

This is the reverse audit companion to Issue #36. Here we document components that ARE specified in the Atlas paper (arXiv:2505.23735) but are either **missing** or **incorrectly implemented** in our code.

### 1. Memory MLP Activation: GELU vs SiLU ❌

**Paper (Section E, chunk 45):**
> "For the memory architecture, unless state otherwise, we use an MLP with 2 layers with expansion factor of 4 and **GELU activation function**"

**Our code (`src/model/atlas_memory.py`):**
```python
# Line 12: "σ is SiLU (Swish) activation"
# Line 94: gate = F.silu(self.w2(x))
# Line 337: gate = F.silu(self.w2(x_mlp_in))
```

**Impact:** GELU and SiLU have different gradients and saturation behaviors. This is a paper fidelity issue.

**Fix:** Change `F.silu()` to `F.gelu()` in `atlas_memory.py`

---

### 2. MAG/MAL Hybrid: Layer-Level Interleaving vs Per-Block Blending ❌

**Paper (Section 5.1, chunk 19):**
> "We also follow Behrouz, Zhong, et al. (2024) and use **two hybrid variants of MAL and MAG** for our Atlas model."

MAG and MAL are **different layer types** interleaved in the architecture:
- **MAG**: Memory Added to Gate (memory output added before attention gate)
- **MAL**: Memory As Last (memory as final layer after attention)

**Our code (`src/model/skeleton.py`):**
```python
# Line 300: Per-block blending gate
self.memory_gate = nn.Parameter(torch.tensor([gate_init]))
# Line ~310: Output blending
gate = torch.sigmoid(self.memory_gate)
mem_out = gate * mem_out
```

We use a **per-block learnable gate** that blends memory into each block. This is NOT the MAG/MAL layer-interleaving pattern.

**Impact:** Different architectural pattern entirely. Paper's MAG/MAL is about which layers use memory vs attention, not per-block blending ratios.

**Note:** This was also flagged in Issue #36 as "NOT in paper". The reverse audit confirms the paper uses layer-level interleaving instead.

---

## Components IN Paper and CORRECT in Code ✓

| Component | Paper Reference | Code Location | Status |
|-----------|-----------------|---------------|--------|
| Short convolutions (size 4) | Section 5.1: "short convolution layers with size 4" | `src/model/projections.py:128` | ✓ |
| K/Q normalization | Section 5.1: "normalization on keys and queries" | `src/model/projections.py` qk_norm | ✓ |
| Newton-Schulz iterations | Section 5: "NewtonShulz5" (5+ iterations) | `ns_iterations: 10` in config | ✓ |
| Polynomial feature mapping | Propositions 1 & 2 | `AtlasMemoryPoly._polynomial_features()` | ✓ |
| Gated MLP structure (Atlas++) | Section 5.1: `M(x) = x + W1(σ(W2x) ⊙ W3x)` | `AtlasMemory` (line 9) | ✓ |
| 2-layer MLP with residual | Section E: "2 layers with residual connections" | `AtlasMemory.forward()` | ✓ |
| Input-dependent γ gates | Equation 9 (Omega Rule) | `GammaGate` class | ✓ |
| LayerNorm at end of chunk | Section E: "layer norm at end of each chunk" | `output_norm` in `AtlasMemoryPoly` | ✓ |

---

### Priority 1: Fix Activation Function
```python
# In src/model/atlas_memory.py, change:
gate = F.silu(self.w2(x))  # Line 94, 337
# To:
gate = F.gelu(self.w2(x))
```

### Priority 2: Clarify MAG/MAL vs Our Approach
Either:
- **Option A:** Implement true MAG/MAL layer interleaving (architectural change)
- **Option B:** Document that our per-block blending is an experimental variant (not paper-faithful)

---

## Related Issues
- Issue #36: Components NOT in paper that should be REMOVED
